src/widgets/RealtimeEffectPanel.cpp

```diff
@@ -0,0 +1,711 @@
+            {
+               currentSubMenu->Bind(wxEVT_MENU, submenuEventHandler);
+               menu.AppendSubMenu(currentSubMenu.release(), currentSubMenuName);
+            }
+            currentSubMenuName = effect.GetVendor();
+            currentSubMenu = std::make_unique<wxMenu>();
+         }
+
+         const auto ID = wxID_HIGHEST + i;
+         currentSubMenu->Append(ID, effect.GetSymbol().Translation());
+      }
+
+      menu.AppendSeparator();
+
+      menu.Append(wxID_MORE, XO("Get more effects...").Translation());
+
+      menu.Bind(wxEVT_MENU, [&](wxCommandEvent& event)
+      {
+         if(event.GetId() == wxID_REMOVE)
+            currentEffectControl->RemoveFromList();
+         else if(event.GetId() == wxID_MORE)
+         {
+            //TODO: implement
+         }
+      });
+
+      if(parent->PopupMenu(&menu) && selectedEffectIndex != -1)
+         return effects[selectedEffectIndex]->GetID();
+      
+      return {};
+   }
+}
+
+class RealtimeEffectListWindow : public wxScrolledWindow
+{
+   std::shared_ptr<Track> mTrack;
+   wxButton* mAddEffect{nullptr};
+   wxWindow* mEffectListContainer{nullptr};
+
+public:
+   RealtimeEffectListWindow(wxWindow *parent,
+                     wxWindowID winid = wxID_ANY,
+                     const wxPoint& pos = wxDefaultPosition,
+                     const wxSize& size = wxDefaultSize,
+                     long style = wxScrolledWindowStyle,
+                     const wxString& name = wxPanelNameStr)
+      : wxScrolledWindow(parent, winid, pos, size, style, name)
+   {
+      #ifdef __WXMSW__
+      //Fixes flickering on redraw
+      SetDoubleBuffered(true);
+   #endif
+
+      //indicates the insertion position of the item
+      auto line = safenew ThemedWindowWrapper<wxWindow>(this, wxID_ANY);
+      line->SetBackgroundColorIndex(clrDropHintHighlight);
+      line->Hide();
+
+      auto rootSizer = std::make_unique<wxBoxSizer>(wxVERTICAL);
+
+      auto addEffect = safenew wxButton(this, wxID_ANY, XO("Add new effect").Translation());
+      addEffect->SetMinSize(wxSize(180, 30));
+      addEffect->Bind(wxEVT_BUTTON, &RealtimeEffectListWindow::OnAddEffectClicked, this);
+      mAddEffect = addEffect;
+
+      auto effectListContainer = safenew ThemedWindowWrapper<wxWindow>(this, wxID_ANY);
+      effectListContainer->SetBackgroundColorIndex(clrMedium);
+      effectListContainer->SetSizer(safenew wxBoxSizer(wxVERTICAL));
+      effectListContainer->SetDoubleBuffered(true);
+      mEffectListContainer = effectListContainer;
+
+      rootSizer->Add(mEffectListContainer, 0, wxEXPAND);
+      rootSizer->Add(addEffect, 0, wxALIGN_CENTER | wxALL, 10);
+
+      SetSizer(rootSizer.release());
+
+      mEffectListContainer->Bind(EVT_EFFECT_ITEM_REMOVE, [this](wxCommandEvent&)
+      {
+         wxWindowUpdateLocker freeze(this);
+         Layout();
+      });
+
+      Bind(EVT_MOVABLE_CONTROL_DRAG_STARTED, [line](const MovableControlEvent& event)
```

Comment:
For these other event types, closer coupling of this window and the items is proper.